packages/xo-web/src/common/utils.js

```diff
@@ -371,13 +370,20 @@ export const resolveResourceSets = resourceSets =>
 //   id: 42,
 // }, 32)
 //
-// newString === 'foo_COPY_foo_42_32'
+// newString === 'foo_COPY_{name}_42_32%'
 // ```
 export function buildTemplate(pattern, rules) {
-  const regExp = new RegExp(join(map(keys(rules), escapeRegExp), '|'), 'g')
+  const transforms = { '\\\\': '\\' }
+  const matches = ['\\\\']
+  forOwn(rules, (output, input) => {
+    transforms[input] = output
+    transforms[`\\${input}`] = input
+    matches.push(input, `\\${input}`)
+  })
+  const regExp = new RegExp(matches.map(escapeRegExp).join('|'), 'g')
```

Comment:
We need to sort matches by descending length or we might have issues:

```js
'foobar'.replace(/foo|foobar/, 'baz')
// → 'bazbar'
```

You can use this comparator:
```js
const compareLengthDesc = (a, b) => b.length - a.length
```